### PR TITLE
Add HTTP API transport.

### PR DESCRIPTION
**Pull request by ptondereau**
- [x] Added Http API transport 

There is a problem with code coverage action and I can't resolve myself but tests pass:

```
2020-06-04 15:44:29.598Z error [CodacyCoverageReporter] Invalid configuration: Empty argument for --project-token  - (CodacyCoverageReporter.scala:25)
```

### CODE REVIEW

**Comment by ptondereau** [comment]
@drixs6o9 any news?

**Comment by drixs6o9** [comment]
> @drixs6o9 any news?

Sorry, didn't see before. I've got some personal problems.
I'll go to check :)

File: README.md [hunk 1]
```diff
@@ -45,6 +45,15 @@ Exemple:
 MAILER_DSN=sendinblue+smtps://$SENDINBLUE_USERNAME:$SENDINBLUE_PASSWORD@default
 ```
 
+**With API**:
```
Comment (drixs6o9) [inline README.md:48]: Please, just add **With HTTP** for the other part :)

File: composer.json [hunk 2]
```diff
@@ -13,6 +13,8 @@
     ],
     "require": {
         "php": "^7.2.9",
+        "symfony/config": "^4.4|^5.0",
+        "symfony/dependency-injection": "^4.4|^5.0",
```
Comment (drixs6o9) [inline composer.json:17]: Why did you add those ? Their are useless :)

You must add **symfony/http-client** like google-mailer please : https://github.com/symfony/google-mailer/blob/master/composer.json

File: Transport/SendinBlueApiTransport.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            if ($address->getName()) {
+                $stringified['name'] = $address->getName();
+            }
+
+            return $stringified;
+        };
+
+        $payload = [
+            'sender' => $addressStringifier($envelope->getSender()),
+            "subject" => $email->getSubject()
```
Comment (drixs6o9) [inline Transport/SendinBlueApiTransport.php:67]:
```suggestion
            'subject' => $email->getSubject()
```

File: composer.json (hunk 2 above)
Comment (ptondereau) [reply composer.json:17]: Because when you want to add feature in your package, my IDE throw missing package error in folder `DependecyInjection`.
They are not useless here!

Comment (ptondereau) [reply composer.json:17]: BTW https://github.com/symfony/google-mailer don't have this dependency because its doesnt use DepencyInjection

**Comment by ptondereau** [comment]
@drixs6o9 all fixed

File: composer.json (hunk 2 above)
Comment (drixs6o9) [reply composer.json:17]: Wich IDE are you using and where is your **DependecyInjection** error ?
Mine doesn't say anything because they are already specified if you follow this package dependencies :
**symfony/mailer -> symfony/mime -> symfony/dependency-injection -> symfony/config**

Then if you look in any ***-mailer** bridge's composer.json of Symfony repositories, there are never mentionned.
Because for me, this bridge is just a part of symfony/mailer and it's to this package to get the responsibility of require them.

File: Transport/SendinblueTransportFactory.php [hunk 4]
```diff
@@ -35,6 +34,13 @@ public function create(Dsn $dsn): TransportInterface
             case 'sendinblue+smtps':
                 $transport = sprintf(self::NAMESPACE, 'SendinblueSmtpsTransport');
                 break;
+            case 'sendinblue+api':
+                $key = $this->getUser($dsn);
+                $host = 'default' === $dsn->getHost() ? null : $dsn->getHost();
+                $port = $dsn->getPort();
+                return (new SendinBlueApiTransport($key, $this->client, $this->dispatcher, $this->logger))
+                    ->setHost($host)->setPort($port);
+                break;
```
Comment (drixs6o9) [inline Transport/SendinblueTransportFactory.php:43]:
```suggestion
```
You don't need to `break;` return is enough :)

File: Transport/SendinBlueApiTransport.php [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+use Symfony\Component\Mailer\Exception\HttpTransportException;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Component\Mailer\Transport\AbstractApiTransport;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Email;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+class SendinblueApiTransport extends AbstractApiTransport
```
Comment (drixs6o9) [inline Transport/SendinBlueApiTransport.php:16]: You've forgot to rename the file :)

Comment (ptondereau) [reply Transport/SendinBlueApiTransport.php:16]: You asked for this change: 
image

Comment (drixs6o9) [reply Transport/SendinBlueApiTransport.php:16]: Sorry, it's a habit to follow PSR-4 and it seemed obvious to me.
https://www.php-fig.org/psr/psr-4/

Comment (ptondereau) [reply Transport/SendinBlueApiTransport.php:16]: Should I keep like this or revert?

**Comment by ptondereau** [comment]
@drixs6o9 last changes fixed

**Comment by drixs6o9** [comment]
Can I push on your PR for helping you ? PR quality review is not passing and i saw an other thing.

**Comment by ptondereau** [comment]
Yes you can

**Comment by drixs6o9** [comment]
I can merge it and publish a new release if it's ok for you. I'll improve the code later.

**Comment by ptondereau** [comment]
@drixs6o9 sure! thank you
